android/app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java

```diff
@@ -80,24 +82,49 @@ public void accept(@NonNull ArrayList<Integer> ids) throws Exception {
                                 bookmarkedIds = ids;
 
                                 sessionsListAdapter.clear();
-                                for (int i = 0; i < bookmarkedIds.size(); i++) {
-                                    Integer id = bookmarkedIds.get(i);
 
-                                    final int index = i;
-                                    dbSingleton.getSessionByIdObservable(id)
-                                            .subscribe(new Consumer<Session>() {
-                                                @Override
-                                                public void accept(@NonNull Session session) throws Exception {
-                                                    mSessions.add(session);
+                                if(bookmarkedIds.size()>0) {
 
-                                                    sessionsListAdapter.notifyItemInserted(index);
+                                    for (int i = 0; i < bookmarkedIds.size(); i++) {
```

Comment:
isn't there a simpler way to do this?
I think that refreshing the adapter that holds bookmarks should be more than enough.